geotrek/common/views.py

```diff
@@ -316,6 +325,87 @@ def get(request, *args, **kwargs):
         return JsonResponse(response)
 
 
+class HDViewPointAPIViewSet(APIViewSet):
+    model = HDViewPoint
+    serializer_class = HDViewPointAPISerializer
+    geojson_serializer_class = HDViewPointAPIGeoJSONSerializer
+
+    def get_queryset(self):
+        return HDViewPoint.objects.annotate(api_geom=Transform("geom", settings.API_SRID))
+
+
+class HDViewPointDetail(CompletenessMixin, mapentity_views.MapEntityDetail):
+    model = HDViewPoint
+    queryset = HDViewPoint.objects.all().select_related('content_type', 'license')
+
+    @method_decorator(login_required)
+    def dispatch(self, *args, **kwargs):
+        return super().dispatch(*args, **kwargs)
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context['can_edit'] = self.get_object().content_object.same_structure(self.request.user)
+        return context
+
+
+class HDViewPointCreate(mapentity_views.MapEntityCreate):
+    model = HDViewPoint
+    form_class = HDViewPointForm
+
+    @method_decorator(login_required)
+    def dispatch(self, *args, **kwargs):
+        return super().dispatch(*args, **kwargs)
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['content_type'] = self.request.GET.get('content_type')
+        kwargs['object_id'] = self.request.GET.get('object_id')
+        return kwargs
+
+
+class HDViewPointUpdate(mapentity_views.MapEntityUpdate):
+    queryset = HDViewPoint.objects.all()
+    form_class = HDViewPointForm
+
+    @method_decorator(login_required)
+    def dispatch(self, *args, **kwargs):
+        return super().dispatch(*args, **kwargs)
+
+
+class HDViewPointDelete(mapentity_views.MapEntityDelete):
+    model = HDViewPoint
+
+    @method_decorator(login_required)
+    def dispatch(self, *args, **kwargs):
+        return super().dispatch(*args, **kwargs)
+
+    def get_success_url(self):
+        return self.get_object().content_object.get_detail_url()
+
+
+class HDViewPointAnnotate(UpdateView):
+    model = HDViewPoint
+    form_class = HDViewPointAnnotationForm
+    template_name_suffix = '_annotation_form'
+
+    @method_decorator(login_required)
+    def dispatch(self, *args, **kwargs):
+        return super().dispatch(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['title'] = self.get_object().title
```

Comment:
as object is by default already passed to context, why pass title and not just use {{ object.title }} in template ?